Source/WebCore/rendering/RenderThemeCocoa.h

```diff
@@ -25,22 +25,73 @@
 
 #pragma once
 
+#include "FloatRoundedRect.h"
+#include "GraphicsContextCG.h"
+#include "Icon.h"
 #include "RenderTheme.h"
+#include <CoreGraphics/CoreGraphics.h>
 #include <wtf/RetainPtr.h>
 
 OBJC_CLASS NSDateComponentsFormatter;
 
 namespace WebCore {
+const float ControlBaseHeight = 20;
+const float ControlBaseFontSize = 11;
+
+struct IOSGradient {
+    float* start; // points to static float[4]
+    float* end; // points to static float[4]
+    IOSGradient(float start[4], float end[4])
+        : start(start)
+        , end(end)
+    {
+    }
+};
+
+typedef IOSGradient* IOSGradientRef;
+
+enum Interpolation {
+    LinearInterpolation,
+    ExponentialInterpolation
+};
+
+enum IOSGradientType {
+    InsetGradient,
+    ShineGradient,
+    ShadeGradient,
+    ConvexGradient,
+    ConcaveGradient,
+    SliderTrackGradient,
+    ReadonlySliderTrackGradient,
+    SliderThumbOpaquePressedGradient,
+};
 
 class RenderThemeCocoa : public RenderTheme {
 public:
     WEBCORE_EXPORT static RenderThemeCocoa& singleton();
 
+protected:
+    static IOSGradientRef gradientWithName(IOSGradientType);
+
+    static CGFunctionRef getSharedFunctionRef(IOSGradientRef, Interpolation);
+
+    static void drawAxialGradient(CGContextRef, IOSGradientRef, const FloatPoint&, const FloatPoint&, Interpolation);
+
```

Comment:
This paragraphing is pretty loose.

Reply:
as In I should remove the newlines between each function?

Reply:
Or put them in sensible groups